plugins/formal-verification/formver.plugin/src/org/jetbrains/kotlin/formver/reporting/VerifierErrorInterpreter.kt

```diff
@@ -40,6 +40,9 @@ class VerifierErrorInterpreter {
             is SourceRole.ParamFunctionLeakageCheck -> with(role) {
                 reportOn(source, PluginErrors.LAMBDA_MAY_LEAK, error.reason.fetchLeakingFunction(), context)
             }
+            is SourceRole.ListElementAccessCheck -> with(role) {
+                reportOn(source, PluginErrors.POSSIBLE_INDEX_OUT_OF_BOUND, error.fetchOutOfBoundsIndexList(), context)
```

Comment:
With this change, there are four different ways that information is extracted:
1. `role.asUserFriendlyMessage` and other `role` properties
2. `role.member.describe()` as a function
3. `error.reason.fetchX()`
4. `error.fetchX()`

It would be nice to either have a unified approach, or at least a clear understanding of why these different approaches are needed and which one works in which case.